### PR TITLE
Adds ability to define testing area

### PR DESCRIPTION
**Pull request by GlynnPhillips**
Allows you to test a part of the page and ignore everything outside of it.

I wanted to raise this PR to get some feedback on the naming before tidying up and putting in the tests.

I was struggling with the option name and I am not sure how people feel about `selector` ?

### CODE REVIEW

**Comment by rowanmanning** [comment]
I like it! Not sure about the name `selector` either, but struggling to think of an alternative right now. Maybe something like `rootElement`?

**Comment by GlynnPhillips** [comment]
Ok I will work on this over the weekend and we can have a think and change it once I have everything worked out and tests written.

**Comment by rowanmanning** [comment]
Cool :) don't work too hard!

**Comment by GlynnPhillips** [comment]
I have now added some tests and documentation. Happy for anyone to correct my documentation if they have a better way of explaining the feature.

**Comment by GlynnPhillips** [comment]
Thanks for the help with documentation @rowanmanning I have made those changes

File: lib/inject.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ function injectPa11y(window, options, done) {
 	}
 
 	function processMessages(messages) {
-		return messages.map(processMessage).filter(isMessageWanted);
+		return messages.filter(isMessageInTestArea).map(processMessage).filter(isMessageWanted);
```
Comment (rowanmanning) [inline lib/inject.js]: Oh actually one more small tweak. Currently if there is no `rootElement` specified, we run `querySelector` on `null`. I wonder whether this function would be more optimal if it was:

```js
function processMessages(messages) {
	if (options.rootElement) {
		messages = messages.filter(isMessageInTestArea);
	}
	return messages.map(processMessage).filter(isMessageWanted);
}
```

**Comment by rowanmanning** [comment]
Sorry, should have spotted that last comment earlier. Otherwise really happy with this :) feel free to merge this and #136 tomorrow and release as `3.6.0`!

File: lib/inject.js (hunk 1 above)
Comment (GlynnPhillips) [inline lib/inject.js]: DOH! yeah I moved that when debugging and forgot to replace it. Good catch

**Pull request merged**